route/build/src/histVars_data.f90

```diff
@@ -260,4 +275,184 @@ SUBROUTINE clean(this)
+      type(io_desc_t)                    :: ioDescHruHist         ! PIO domain decomposition data for hru runoff [nHRU]
+
+      ierr=0; message='read_restart/'
+
+      call openFile(pioSystem, pioFileDesc, trim(restart_name), pio_typename, ncd_nowrite, FileStatus, ierr, cmessage)
+      if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+
+      call get_nc(trim(restart_name), 'history_time', this%timeVar, 1, ierr, cmessage) ! get restart datetime
+      call get_nc(trim(restart_name), 'nt', this%nt, 1, ierr, cmessage)                ! get number of aggretation time steps
+
+      ! get number of HRUs and Reaches for each core
+      if (masterproc) then
+        this%nHru =nHru_mainstem+nHru_trib
+        this%nRch =nRch_mainstem+nRch_trib+nTribOutlet
+        allocate(array_tmp(nRch_mainstem+nRch_trib))
+      else
+        this%nHru = nHru_trib
+        this%nRch = nRch_trib
+        allocate(array_tmp(nRch_trib))
+      end if
+
+      ! global reach and hru index
+      if (masterproc) then
+        allocate(compdof_rch(sum(rch_per_proc(-1:pid))))
+        allocate(compdof_hru(sum(hru_per_proc(-1:pid))))
+      else
+        allocate(compdof_rch(rch_per_proc(pid)))
+        allocate(compdof_hru(hru_per_proc(pid)))
+      end if
+
+      ! For reach flux/volume
+      if (masterproc) then
+        ix1 = 1
+      else
+        ix1 = sum(rch_per_proc(-1:pid-1))+1
+      endif
+      ix2 = sum(rch_per_proc(-1:pid))
+      compdof_rch = arth(ix1, 1, ix2-ix1+1)
+
+      ! For HRU flux/volume
+      if (masterproc) then
+        ix1 = 1
+      else
+        ix1 = sum(hru_per_proc(-1:pid-1))+1
+      endif
+      ix2 = sum(hru_per_proc(-1:pid))
+      compdof_hru = arth(ix1, 1, ix2-ix1+1)
+
+      ! initialze ioDesc
+      call pio_decomp(pioSystem,         & ! input: pio system descriptor
+                      ncd_float,         & ! input: data type (pio_int, pio_real, pio_double, pio_char)
+                      [nRch],            & ! input: dimension length == global array size
+                      compdof_rch,       & ! input: local->global mapping
+                      ioDescRchHist)
+
+      call pio_decomp(pioSystem,         & ! input: pio system descriptor
+                      ncd_float,         & ! input: data type (pio_int, pio_real, pio_double, pio_char)
+                      [nHRU],            & ! input: dimension length == global array size
+                      compdof_hru,       & ! input: local->global mapping
+                      ioDescHruHist)
+
+      ! reading basin runoff (hru dimension)
+      if (meta_hflx(ixHFLX%basRunoff)%varFile) then
+        allocate(this%basRunoff(this%nHru), stat=ierr, errmsg=cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage)//' [hVars%basRunoff]'; return; endif
+
+        call read_pnetcdf(pioFileDesc, meta_hflx(ixHFLX%basRunoff)%varName, this%basRunoff, ioDescHruHist, ierr, cmessage)
```

Comment:
The name read_pnetcdf makes it sound like it's using PNETCDF rather than PIO. It might be good to change the name to avoid that confusion.

Reply:
changed name from `read_pnetcdf` to `read_dist_array`.